core/dbt/parser/schemas.py

```diff
@@ -827,8 +886,11 @@ def parse_patch(
                     # re-application of the patch in partial parsing.
                     node.patch_path = source_file.file_id
             else:
-                # Should we issue a warning message here?
-                return
+                raise ParsingException(
+                    f"Did not find matching node for patch with name '{patch.name}' "
+                    f"in the '{patch.yaml_key}' section of "
+                    f"file '{source_file.path.original_file_path}'"
+                )
```

Comment:
@gshank This is on me for not catching before we merged. This is going to be a breaking change for many users. Historically, we've supported the ability to define patches for nodes that don't exist; they simply aren't applied.

Any chance we could switch this from an exception to a debug-level warning message, and keep going?

Reply:
Yes, that would be fine. We still create the tests for non-existent nodes, which seems kind of wrong, but it won't affect the env_var parsing.

Reply:
Agree, that does seem sort of wrong. Those tests will exist in the manifest, but I don't think they'll actually be executed. For later :shrug: